### PR TITLE
cmake-build: remove gtk3-dependency from wxbase

### PR DESCRIPTION
**Pull request by jolz**
libwx_baseu_3.1.so should not include "-lgtk-3" during link. 
This change has only been tested with linux+gtk2/3 (sorry.)

### CODE REVIEW

**Comment by vadz** [comment]
Thanks for the fix, this seems correct to me, but, as always, I'd appreciate a confirmation from @MaartenBent before merging (TIA!).

**Review by MaartenBent — CHANGES_REQUESTED** [review]
Maybe we should add a similar check with `if(wxTOOLKIT_INCLUDE_DIRS AND NOT is_base)`?

We always want `wxTOOLKIT_DEFINITIONS` so nothing should change there.

File: build/cmake/functions.cmake
Comment (jolz) [reply build/cmake/functions.cmake]: Yes, you are right. 
i've just tested it and it works for linux/gtk2 and linux/gtk3.

I'm not sure why we need `-D__WXGTK__`, but <wx/defs.h> requests it (as you said). 

(sorry, this is the first time that i use "pull requests". i hope it stays readable.)

**Comment by vadz** [comment]
Thanks for the update!

I'm going to merge this soon if there are no objections.

**Comment by vadz** [comment]
Merged now, thanks!

**Pull request closed without merging**
